trio/_core/_run.py

```diff
@@ -510,7 +529,17 @@ def _close(self, exc):
             and self._cancel_status.effectively_cancelled
             and not self._cancel_status.parent_cancellation_is_visible_to_us
         ):
-            exc = MultiError.filter(self._exc_filter, exc)
+            if isinstance(exc, Cancelled):
```

Comment:
The new behavior here has a number of differences from the old, which I'm not sure were intended and which I've pointed out individually. Do you think it might be wise to just use the old `MultiError.filter()` instead of trying to reimplement its semantics imprecisely? You could expose it under an internal non-deprecated name. And then when everyone has ported over to ExceptionGroups we can remove MultiError as well as the collapsing of single-error MultiErrors into their constituent.

Reply:
I don't think you addressed any of these comments (this one or the ones in `collapse_exception_group()` that it's referring to).

Reply:
> The new behavior here has a number of differences from the old

What differences are you referring to? We now have the `strict_exception_groups` option which should be respected, and using `MultiError.filter()` would unconditionally flatten the group.

Reply:
The differences/issues I'm still tracking are the traceback concatenation (which you're now doing, but in the wrong order) and the extra collapsing added by a cancel scope around a strict_exception_groups nursery. I commented on both of those above. I think I originally had a third comment but removed it because I realized it was mistaken; sorry for the confusion.